### PR TITLE
[ISSUE #4136] Fix the bug that cann't correctlly instantiate ServiceInfo from cache file

### PR DESCRIPTION
**Pull request by shizhengxing**
Please do not create a Pull Request without creating an issue first.
#4136 
#4183 

## What is the purpose of the change
按照 #4183 中讨论的要求重新修改提交

### CODE REVIEW

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java [hunk 1]
```diff
@@ -197,11 +217,14 @@ public static String getKey(String name, String clusters) {
...
-            return getKey(URLEncoder.encode(name, "UTF-8"), clusters);
-        } catch (UnsupportedEncodingException e) {
-            return getKey();
-        }
+				String serviceName = getGroupedServiceName();
+				try {
+						serviceName = URLEncoder.encode(serviceName, "UTF-8");
+				} catch (UnsupportedEncodingException e) {
+				}
+
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java:225]: remove unused blank line. Thanks.

File: api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java [hunk 2]
```diff
@@ -197,11 +217,14 @@ public static String getKey(String name, String clusters) {
...
-            return getKey();
-        }
+				String serviceName = getGroupedServiceName();
+				try {
+						serviceName = URLEncoder.encode(serviceName, "UTF-8");
+				} catch (UnsupportedEncodingException e) {
+				}
+		
+				String key = getKey(serviceName, clusters);
+				return key;
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/pojo/ServiceInfo.java:227]: can return getKey() directly. do not define a variable if no use it.

File: api/src/test/java/com/alibaba/nacos/api/naming/pojo/ServiceInfoTest.java [hunk 3]
```diff
@@ -21,30 +21,29 @@
...
+import java.net.URLEncoder;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
 
 public class ServiceInfoTest {
-    
+
```
Comment (KomachiSion) [inline api/src/test/java/com/alibaba/nacos/api/naming/pojo/ServiceInfoTest.java:31]: Please use nacos code style to reformat. 

you can see relative information in source code `style/codeStyle.md`

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
@shizhengxing The CI failed, error is check style can't pass, please take some time to solve it. detail see source code `style/codeStyle.md`. 
Thanks.

**Review by KomachiSion — CHANGES_REQUESTED** [review]
Please use load nacos code style to reformat your changes.

If the code style and check style can't pass, we will refuse this PR.

**Comment by shizhengxing** [comment]
> Please use load nacos code style to reformat your changes.
> 
> If the code style and check style can't pass, we will refuse this PR.

progressing,gives me more time

**Comment by shizhengxing** [comment]
@KomachiSion I only changed three files,bug after fetch from nacos repository,It turns as 7 changes!
Will I need close this PR,and open a new one ?I am unfamiliar with how two deal this situation!

File: config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedPermissionPersistServiceImpl.java [hunk 4]
```diff
@@ -51,15 +53,17 @@
         
         String sqlCountRows = "select count(*) from permissions where ";
         String sqlFetchRows = "select role,resource,action from permissions where ";
-        
-        String where = " role='" + role + "' ";
-        
-        if (StringUtils.isBlank(role)) {
+    
+        String where = " role= ? ";
+        List<String> params = new ArrayList<>();
```
Comment (sanxun0325) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedPermissionPersistServiceImpl.java:58]: Params might be better defined directly as arrays

File: config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedRolePersistServiceImpl.java [hunk 5]
```diff
@@ -74,14 +75,16 @@
         
         String sqlCountRows = "select count(*) from roles where ";
         String sqlFetchRows = "select role,username from roles where ";
-        
-        String where = " username='" + username + "' ";
-        
-        if (StringUtils.isBlank(username)) {
+    
+        String where = " username= ? ";
+        List<String> params = new ArrayList<>();
```
Comment (sanxun0325) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedRolePersistServiceImpl.java:80]: Params might be better defined directly as arrays

File: config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalRolePersistServiceImpl.java [hunk 6]
```diff
@@ -88,16 +89,17 @@ protected void init() {
         String sqlCountRows = "select count(*) from roles where ";
         String sqlFetchRows = "select role,username from roles where ";
         
-        String where = " username='" + username + "' ";
-        
-        if (StringUtils.isBlank(username)) {
+        String where = " username= ? ";
+        List<String> params = new ArrayList<>();
```
Comment (sanxun0325) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalRolePersistServiceImpl.java:93]: Params might be better defined directly as arrays

File: config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalPermissionPersistServiceImpl.java [hunk 7]
```diff
@@ -57,16 +59,18 @@ protected void init() {
         
         String sqlCountRows = "select count(*) from permissions where ";
         String sqlFetchRows = "select role,resource,action from permissions where ";
-        
-        String where = " role='" + role + "' ";
-        
-        if (StringUtils.isBlank(role)) {
+    
+        String where = " role= ? ";
+        List<String> params = new ArrayList<>();
```
Comment (sanxun0325) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalPermissionPersistServiceImpl.java:64]: Params might be better defined directly as arrays

**Comment by KomachiSion** [comment]
> @KomachiSion I only changed three files,but after fetch from nacos repository,It turns as 7 changes!
> Will I need to close this PR,and open a new one ?I am unfamiliar with how to deal with this situation
> I need your help!

I found that you merge other change to your PR, so the files increase, If you can use rebase to fix it , you do not need close PR. If you don't know how to use rebase to fix, the best way is checkout a new branch and re-submit PR.

At the next time, Please use nacos code style to reformat so that it can be merged quickly.

**Comment by shizhengxing** [comment]
> > @KomachiSion I only changed three files,but after fetch from nacos repository,It turns as 7 changes!
> > Will I need to close this PR,and open a new one ?I am unfamiliar with how to deal with this situation
> > I need your help!
> 
> I found that you merge other change to your PR, so the files increase, If you can use rebase to fix it , you do not need close PR. If you don't know how to use rebase to fix, the best way is checkout a new branch and re-submit PR.
> 
> At the next time, Please use nacos code style to reformat so that it can be merged quickly.

OK thanks,I am learning how to use rebase,and I will solve this today.
What's more,I have already learned how to use checkstyle last night,and i was already used it to format this PR's submit.So the CI passed.Thanks!

**Pull request closed without merging**